### PR TITLE
Add queryable state store APIs and infrastructure.

### PR DESCRIPTION
**Pull request by philliphoff**
Add the state store APIs and infrastructure related to queries.

### CODE REVIEW

File: src/Dapr.PluggableComponents.AspNetCore/DaprPluggableComponentsServiceBuilder.cs [hunk 1]
```diff
@@ -219,4 +223,12 @@ public DaprPluggableComponentsServiceBuilder RegisterStateStore<TStateStore>(Com
...
         }
     }
+
+    private void AddRelatedStateStoreServices<TStateStore>() where TStateStore : class
+    {
+        if (typeof(TStateStore).IsAssignableTo(typeof(IQueryableStateStore)))
+        {
+            this.AddRelatedService<IQueryableStateStore, TStateStore, QueryableStateStoreAdaptor>();
+        }
+    }
```
Comment (tmacam) [inline src/Dapr.PluggableComponents.AspNetCore/DaprPluggableComponentsServiceBuilder.cs:233]: This will conflict with #9, right?

Comment (tmacam) [inline src/Dapr.PluggableComponents.AspNetCore/DaprPluggableComponentsServiceBuilder.cs:233]: This will conflict with PR #9, right?

**Review by tmacam — APPROVED** [review]
LGTM

**Review by mcandeia — APPROVED** [review]

File: src/Dapr.PluggableComponents.AspNetCore/DaprPluggableComponentsServiceBuilder.cs (hunk 1 above)
Comment (philliphoff) [reply src/Dapr.PluggableComponents.AspNetCore/DaprPluggableComponentsServiceBuilder.cs:233]: One of them will win, yes, and I'll have to merge/rebase/resolve the other.

**Pull request merged**